arch/arc/kernel/perf_event.c

```diff
@@ -113,8 +113,12 @@ static u64 arc_pmu_read_counter(int idx)
 	write_aux_reg(ARC_REG_PCT_INDEX, idx);
 	tmp = read_aux_reg(ARC_REG_PCT_CONTROL);
 	write_aux_reg(ARC_REG_PCT_CONTROL, tmp | ARC_REG_PCT_CONTROL_SN);
+#ifdef CONFIG_ISA_ARCV3
```

Comment:
1. Please try to use IS_ENABLED() where possible to avoid littering #ifdef'ery
2. AFAIK ARCv3:32 won't have 64-bit so this will break- see 6483bc4f30c2af8515ec4fa5b3a1737c4bb0692c for CONFIG_ISA_ARCV3 vs. CONFIG_64BIT

Reply:
also FWIW, IS_ENABLED() is slightly different semantically than #ifdef since the code gets compiled (if not used) even if the config symbols is not enabled.